### PR TITLE
Add basic ga tag w/ settable campaignid

### PR DESCRIPTION
**Pull request by scbedd**
Alrighty, I think this was about the cleanest as I could find to do this.

1. Add new header template that adds GA tag.
   - This header was generated by `docfx template export default` -> add my stuff -> copy into partials folder.
2. Add campaignId argument to GenerateDocFxToCConfig (which is really the function that builds the skeleton of the site)
   - This will optionally inject a specific campaignId if presented

Have run a test using the java folks as a guinea pig. To take advantage of this, merely update the call in `Language-Settings.ps1` -> `Get-<language>-GithubIODocIndex()`

image

### CODE REVIEW

File: eng/common/docgeneration/Generate-DocIndex.ps1 [hunk 1]
```diff
@@ -87,10 +87,10 @@ function Get-TocMapping {
         }
         $orderServiceMapping[$artifact] = @($serviceName, $displayName)
     }
-    return $orderServiceMapping                   
+    return $orderServiceMapping
 }
 
-function GenerateDocfxTocContent([Hashtable]$tocContent, [String]$lang) {
+function GenerateDocfxTocContent([Hashtable]$tocContent, [String]$lang, [String]$campaignId = "UA-62780441-46") {
```
Comment (scbedd) [inline eng/common/docgeneration/Generate-DocIndex.ps1:93]: This "default" is the `general` campaign Id.

File: eng/common/docgeneration/templates/matthews/partials/head.tmpl.partial [hunk 2]
```diff
@@ -0,0 +1,28 @@
+<head>
```
Comment (scbedd) [inline eng/common/docgeneration/templates/matthews/partials/head.tmpl.partial:1]: This is just the `default` head template with a couple adjustments.

File: eng/common/docgeneration/Generate-DocIndex.ps1 (hunk 1 above)
Comment (weshaggard) [reply eng/common/docgeneration/Generate-DocIndex.ps1:93]: Does this value need to be protected?

File: eng/common/docgeneration/templates/matthews/partials/head.tmpl.partial [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  <link rel="stylesheet" href="{{_rel}}styles/docfx.css">
+  <link rel="stylesheet" href="{{_rel}}styles/main.css">
+  <meta property="docfx:navrel" content="{{_navRel}}">
+  <meta property="docfx:tocrel" content="{{_tocRel}}">
+  {{#_noindex}}<meta name="searchOption" content="noindex">{{/_noindex}}
+  {{#_enableSearch}}<meta property="docfx:rel" content="{{_rel}}">{{/_enableSearch}}
+  {{#_enableNewTab}}<meta property="docfx:newtab" content="true">{{/_enableNewTab}}
+
+  <!-- Global site tag (gtag.js) - Google Analytics -->
+  <script async src="https://www.googletagmanager.com/gtag/js?id=GA_CAMPAIGN_ID"></script>
```
Comment (weshaggard) [inline eng/common/docgeneration/templates/matthews/partials/head.tmpl.partial:20]: Is this only in our github.io docs or does it also go on our msdocs?

**Review by chidozieononiwu — APPROVED** [review]

File: eng/common/docgeneration/Generate-DocIndex.ps1 (hunk 1 above)
Comment (scbedd) [reply eng/common/docgeneration/Generate-DocIndex.ps1:93]: No. It's stamped into every website that we serve. No way to keep it private.

File: eng/common/docgeneration/templates/matthews/partials/head.tmpl.partial (hunk 3 above)
Comment (scbedd) [reply eng/common/docgeneration/templates/matthews/partials/head.tmpl.partial:20]: This is only github.io docs.

**Comment by scbedd** [comment]
Going to do a couple template releases, then just merge this. It's a fairly transparent update.

**Review by weshaggard — APPROVED** [review]
Looks reasonable. Did you plan to also add GA support to the azure-sdk github.io site?

**Pull request merged**